client/state/sites/vouchers/actions.js

```diff
@@ -75,6 +79,68 @@ export const vouchersRequestFailureAction = ( siteId, error ) => {
 	};
 };
 
+export const vouchersAssignReceiveAction = ( siteId, serviceType, voucher ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_RECEIVE,
+		siteId,
+		serviceType,
+		voucher
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST action object
+ *
+ * @param {Number} siteId - side identifier
+ * @param {String} serviceType - service type
+ * @return {Object} siteId - action object
+ */
+export const vouchersAssignRequestAction = ( siteId, serviceType ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_REQUEST,
+		siteId,
+		serviceType
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST_SUCCESS action object
+ *
+ * @param {Number} siteId - side identifier
+ * @param {String} serviceType - service type
+ * @return {Object} siteId - action object
+ */
+export const vouchersAssignRequestSuccessAction = ( siteId, serviceType ) => {
+	return {
+		type: SITE_VOUCHERS_ASSIGN_REQUEST_SUCCESS,
+		siteId,
+		serviceType
+	};
+};
+
+/**
+ * Action creator function
+ *
+ * Return SITE_VOUCHERS_ASSIGN_REQUEST_FAILURE action object
+ *
+ * @param {Number} siteId - site identifier
+ * @param {String} serviceType - service type
+ * @param {Object} error - error message according to REST-API error response
+ * @return {Object} action object
+ */
+export const vouchersAssignRequestFailureAction = ( siteId, serviceType, error ) => {
```

Comment:
Sometimes the action functions are imported directly from the file to be used:

```es6
import { requestSiteVouchers } from 'state/sites/vouchers/actions';
````

However we could use an alias on the fly to identify what the functions is doing in the context when it is used

```es6
import { requestVouchers as requestSiteVouchers } from 'state/sites/vouchers/actions';
```

I agree to change the names if you think that this is better to read and understand the code.